skimage/filters/ridges.py

```diff
@@ -286,6 +298,14 @@ def sato(image, sigmas=range(1, 10, 2), black_ridges=True):
     if np.any(sigmas < 0.0):
         raise ValueError('Sigma values less than zero are not valid')
 
+    if mode is None:
+        warn("Previously, sato implicitly used 'constant' as the "
+             "border mode when dealing with the edge of the array. The new "
+             "behavior is 'reflect'. To recover the old behavior, use "
+             "mode='constant'. To avoid this warning, please explicitely "
+             "set the mode.", category=FutureWarning, stacklevel=2)
+        mode = 'reflect'
```

Comment:
Are there any tests where this warning is raised? See `skimage._shared._testing.expected_warnings`.

Reply:
I took care of the warning raise by specifying the mode in the tests ;-)

Reply:
Yes, but you need a test where you check that the warning is correctly raised when no mode is provided. We once got some code in where there was a RuntimeError inside the warning code, because no test covered that code! I think `warn` had not been imported iirc. 😬

Reply:
Done ;)